lib/gollum-lib/filter/sourcemap.rb

```diff
@@ -0,0 +1,24 @@
+# ~*~ encoding: utf-8 ~*~
+
+# Source Maps
+#
+# Find lines corresponding to h1 headers for the present markup and insert markers indicating the line number in the raw source.
+
+class Gollum::Filter::SourceMap < Gollum::Filter
+  def extract(data)
+  	regex = Gollum::Markup.formats[@markup.format][:h1]
+    return data if regex.nil?
+    i = 0
+    data.lines.map do |line|
```

Comment:
Interesting idea @bartkamphorst. Unfortunately the index will be the nth match, and not the line number. But we can work around this by matching *every* line, and only substituting if the line matches the further conditions for an h1, like so:
```ruby
1.9.3-p362 :071 > "a\n# b    f\na".gsub(/^(#\s+)?(\w+.+)?/).with_index do |_, i|
1.9.3-p362 :072 >     "#{$1}GOLLUMSRCMP#{i}GOLLUMSRCMP#{$2}" if $1 && $2
1.9.3-p362 :073?>   end
 => "a\n# GOLLUMSRCMP1GOLLUMSRCMPb    f\na" 
```
It's not very transparent but that may be worth it if it performs significantly better?